test/integration/skaffold_test.go

```diff
@@ -0,0 +1,119 @@
+// +build integration
+
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/docker/machine/libmachine/mcnutils"
+	"github.com/hashicorp/go-getter"
+	"k8s.io/minikube/pkg/util/retry"
+)
+
+func TestSkaffold(t *testing.T) {
+	if NoneDriver() {
+		t.Skip("none driver doesn't support `minikube docker-env`; skaffold depends on this command")
+	}
+	// can't use a unique profile, as skaffold only recognizes the
+	// profile name 'minikube' as a local cluster
+	profile := "minikube"
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(5))
+	defer CleanupWithLogs(t, profile, cancel)
+
+	// install latest skaffold release
+	tf, err := installSkaffold()
+	if err != nil {
+		t.Fatalf("skaffold release installation failed: %v", err)
+	}
+	defer os.Remove(tf.Name())
+
+	// start minikube cluster
+	args := append([]string{"start", "-p", profile, "--memory=2200"}, StartArgs()...)
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	if err != nil {
+		t.Fatalf("starting minikube: %v\n%s", err, rr.Output())
+	}
+
+	// make sure minikube binary is in path so that skaffold can access it
+	abs, err := filepath.Abs(Target())
+	// copy minikube binary to minikube
+	if err := mcnutils.CopyFile(Target(), filepath.Join(filepath.Dir(abs), "minikube")); err != nil {
+		t.Fatalf("error copying to minikube")
+	}
+	if err != nil {
+		t.Fatalf("absolute path to minikube binary: %v", err)
+	}
+	os.Setenv("PATH", fmt.Sprintf("%s:%s", filepath.Dir(abs), os.Getenv("PATH")))
+	// make sure 'docker' and 'minikube' are on PATH
+	for _, binary := range []string{"minikube", "docker"} {
```

Comment:
the minikube binary in our tests is not called minikube, but minikube-os-arch
and some of our CI machines we have a very old minikube installed in the /usr/loca/bin/
this means skaffold will use that binary?

Reply:
minikube-GOOS-amd64 is copied to`minikube` in line 64 of this test

the `minikube` directory is added to the beginning of PATH in line 70 so we know that skaffold will use the correct binary